File: ssTools/netcat/js/nc.js
const net = require("net");

function handleConnection(socket) {
	console.log(`Menerima koneksi dari ${socket.remoteAddress}:${socket.remotePort}`);
	socket.on("data", (data) => {
		console.log(data.toString());
	});

	socket.on("end", () => {
		console.log("Koneksi ditutup");
	});

	socket.on("error", (error) => {
		console.error("Terjadi kesalahan koneksi:", error);
	});
}

const server = net.createServer(handleConnection);

server.listen(6969, () => {
	console.log("Menjalankan server netcat di port 6969...");
});

File: ssTools/netcat/rust/nc.rs
use std::io::Read;
use std::net::{TcpListener, TcpStream};

fn handle_connection(mut stream: TcpStream) {
    println!("Menerima koneksi dari {}", stream.peer_addr().unwrap());

    let mut buffer = [0; 1024];
    loop {
        match stream.read(&mut buffer) {
            Ok(0) => break, // Koneksi ditutup
            Ok(bytes_read) => {
                // Menampilkan data yang diterima
                print!("{}", String::from_utf8_lossy(&buffer[..bytes_read]));
            }
            Err(e) => {
                eprintln!("Terjadi kesalahan saat membaca data: {}", e);
                break;
            }
        }
    }
}

fn main() {
    let listener = TcpListener::bind("127.0.0.1:6969").expect("Gagal mengikat soket ke port 6969");

    println!("Menjalankan server netcat di port 6969...");

    for stream in listener.incoming() {
        match stream {
            Ok(stream) => {
                // Menangani setiap koneksi dalam thread terpisah
                std::thread::spawn(move || {
                    handle_connection(stream);
                });
            }
            Err(e) => {
                eprintln!("Terjadi kesalahan saat menerima koneksi: {}", e);
                break;
            }
        }
    }
}
